tests/build/build_templates.go

```diff
@@ -141,10 +174,21 @@ var _ = framework.BuildSuiteDescribe("Build templates E2E test", Label("build",
 					if !pipelineRun.HasStarted() {
 						return fmt.Errorf("pipelinerun %s/%s has not started yet", pipelineRun.GetNamespace(), pipelineRun.GetName())
 					}
+					err = kubeadminClient.TektonController.AddFinalizerToPipelineRun(pipelineRun, finalizerName)
+					if err != nil {
+						return fmt.Errorf("error while adding finalizer: %v", err)
+					}
 					return nil
 				}, timeout, constants.PipelineRunPollingInterval).Should(Succeed(), fmt.Sprintf("timed out when waiting for the PipelineRun to start for the Component %s/%s", testNamespace, componentNames[i]))
 			})
 
+		}
+
+		for i, gitUrl := range componentUrls {
+			i := i
+			gitUrl := gitUrl
```

Comment:
Are you sure that this is necessary?

Reply:
yes it was already present earlier in the tests. when I tried to remove them the tests were not behaving as expected.. I think we have to update the variables `i` and `gitUrl` separately inside the `for` loop as well.. tried to find the reason in the internet.. got one ginkgo issue where it was [explained](https://github.com/onsi/ginkgo/issues/462#issuecomment-384633172)